src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2TestBase.cs

```diff
@@ -1208,7 +1208,7 @@ internal async Task<Http2FrameWithPayload> ReceiveFrameAsync(uint maxFrameSize =
 
     internal async Task<Http2FrameWithPayload> ExpectAsync(Http2FrameType type, int withLength, byte withFlags, int withStreamId)
     {
-        var frame = await ReceiveFrameAsync((uint)withLength);
+        var frame = await ReceiveFrameAsync();
```

Comment:
I changed this here and the max value because when Expect fails it's usually because it received a different frame than expected, and the frame length is the least interesting bit of information. I want the checks below for the type and flags to run and then it verifies the length last.